### PR TITLE
Require voted entity exists (issue #143)

### PR DESCRIPTION
**Pull request by takaaki7**
This PR is work in progress, compilable but have not yet been tested.
I think this simple validation may prevent voting for non-existent entity. (#143)
~~And also we should check that all votes are unique.~~(`account_options.votes` is a `flat_set` so uniqueness check is unnecessary. )
May I go on with this plan?

### CODE REVIEW

File: libraries/chain/account_evaluator.cpp [hunk 1]
```diff
@@ -88,7 +90,34 @@ void verify_account_votes( const database& db, const account_options& options )
...
    }
-
+   if (true /* db.head_block_time() >= HARDFORK_NEXT_TIME*/ ) {
+      int voted_committee_count = 0;
+      int voted_witness_count = 0;
+      const auto& approve_worker_idx = db.get_index_type<worker_index>().indices().get<by_vote_for>();
+      const auto& committee_idx = db.get_index_type<committee_member_index>().indices().get<by_vote_id>();
+      const auto& witness_idx = db.get_index_type<witness_index>().indices().get<by_vote_id>();
+      for ( auto id : options.votes ) {
+         switch ( id ) {
```
Comment (pmconrad) [inline libraries/chain/account_evaluator.cpp]: I think the switch should operate on id.type()

**Comment by pmconrad** [comment]
Of course you may. Thank for taking up the task!

**Comment by takaaki7** [comment]
I've modified!
- Voted committees/witnesses counts are validated in account.cpp, so not validate here.
- To avoid duplicating hardfork number with hardforks created in previous repository, append "-2" suffix to issue number. 
- Hardfork time is set at 08/28/2017 @ 12:00am (UTC).

Tested by using original func and vote for non-exist entities as follows.
```
signed_transaction vote_by_id(string voting_account,
                                        string vote_id,
                                        bool approve,
                                        bool broadcast /* = false */)
{ try {
      account_object voting_account_object = get_account(voting_account);
      voting_account_object.options.votes.insert(vote_id_type(vote_id));
      account_update_operation account_update_op;
      account_update_op.account = voting_account_object.id;
      account_update_op.new_options = voting_account_object.options;

      signed_transaction tx;
      tx.operations.push_back( account_update_op );
      set_operation_fees( tx, _remote_db->get_global_properties().parameters.current_fees);
      tx.validate();

      return sign_transaction( tx, broadcast );
} FC_CAPTURE_AND_RETHROW( (voting_account)(vote_id)(approve)(broadcast) ) }
```
```
unlocked >>> vote_by_id nathan "0:10" true true
vote_by_id nathan "0:10" true true
889304ms th_a       wallet.cpp:1908               sign_transaction     ] Caught exception while broadcasting tx 075f27c618eefbce33e149c15ea167f791d860c2:  0 exception: unspecified
Assert Exception: committee_idx.find(id) != committee_idx.end():
```

File: libraries/chain/hardfork.d/143-2.hf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// #143 Require voted entities to exist
+#ifndef HARDFORK_143_2_TIME
+#define HARDFORK_143_2_TIME (fc::time_point_sec( 1503878400 ))
```
Comment (pmconrad) [inline libraries/chain/hardfork.d/143-2.hf]: Please rename to HARDFORK_CORE_143_TIME to make it clear that the number refers to the bitshares-core issue.

**Comment by pmconrad** [comment]
Good job, thanks!
Tested with HF date in the future - replays successfully
Tested with HF date before 02/2016 - replay fails
Tested with HF date in June 2016 - replays successfully

**Comment by takaaki7** [comment]
I'm sorry for being too late. I've fixed it.

**Pull request merged**